### PR TITLE
[Enhancement]Error handling inside SwiftConcurrency tasks

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Provide clear error logs for every task that can fail.

### 📝 Summary

SwiftConcurrency tasks swallowing any error that was thrown inside their body outside of an do/catch block ([reference](https://forums.swift.org/t/task-initializer-with-throwing-closure-swallows-error/56066)).

As those errors may produce unwanted behaviour we need to be able at least to track the error. 

### 🛠 Implementation

This PR does 2 things:
- Integrate SwiftLint into our swift format/lint flow. Currently the only rule that's enabled for swiftLint is the [unhandled_throwing_task](https://realm.github.io/SwiftLint/unhandled_throwing_task.html).
- Fixes all errors produced from SwiftLint by wrapping throwing operations inside tasks into simple do/catch blocks with error logging. Future implementations should consider more extensive error handling.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -647,10 +647,12 @@ end
...
 lane :run_swift_format do |options|
   Dir.chdir('..') do
-    action = options[:lint] ? "--lint" : ""
-
+    swiftFormatAction = options[:lint] ? "--lint" : ""
+    swiftLintAction = options[:lint] ? "lint" : "analyze --fix"
+    
     swiftformat_source_paths.each do |path|
-      sh("mint run swiftformat #{action} --config .swiftformat --exclude #{swiftformat_excluded_paths.join(',')} #{path}")
+      sh("mint run swiftformat #{swiftFormatAction} --config .swiftformat --exclude #{swiftformat_excluded_paths.join(',')} #{path}")
```
Comment (martinmitrevski) [inline fastlane/Fastfile:654]: the linter complains about `swiftFormatAction` 😁

**Review by martinmitrevski — APPROVED** [review]
looks good, we just need to fix that lint error

File: fastlane/Fastfile [hunk 2]
```diff
@@ -648,9 +648,9 @@ desc 'Run source code formatting/linting'
 lane :run_swift_format do |options|
```
Comment (testableapple) [inline fastlane/Fastfile:648]: I think we could now rename this lane. But up to you.

**Review by testableapple — APPROVED** [review]
LGTM ✅

File: fastlane/Fastfile (hunk 2 above)
Comment (ipavlidakis) [reply fastlane/Fastfile:648]: As renaming is going to be a no-production-code change, let me merge this one and we can do it in a subsequent PR. what name do you suggest?

**Pull request merged**